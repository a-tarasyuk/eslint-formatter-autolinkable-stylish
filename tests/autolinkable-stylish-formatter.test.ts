import { ESLint } from 'eslint';
import autolinkableStylishFormatter from '../src/autolinkable-stylish-formatter';
import report from './__fixtures__/report.json';
import parsingErrorsReport from './__fixtures__/parsing-errors-report.json';

describe('autolinkable-stylish-formatter', () => {
  it('matches its snapshot', () => {
    expect(
      autolinkableStylishFormatter(report as ESLint.LintResult[]),
    ).toMatchSnapshot();
  });

  it('matches its snapshot with parsing errors', () => {
    expect(
      autolinkableStylishFormatter(parsingErrorsReport as ESLint.LintResult[]),
    ).toMatchSnapshot();
  });
});
