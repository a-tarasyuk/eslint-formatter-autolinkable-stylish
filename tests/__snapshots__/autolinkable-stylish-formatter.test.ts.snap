// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autolinkable-stylish-formatter matches its snapshot 1`] = `
"[37m/fixtures/a.ts[39m
[31mError[39m: /fixtures/a.ts:1:11   [90m@typescript-eslint/interface-name-prefix[39m [33mInterface name must not be prefixed with \\"I\\".[39m
[31mError[39m: /fixtures/a.ts:1:11   [90m@typescript-eslint/no-empty-interface[39m    [33mAn empty interface is equivalent to \`{}\`.[39m
[31mError[39m: /fixtures/a.ts:5:8    [90mno-empty[39m                                 [33mEmpty block statement.[39m
[31mError[39m: /fixtures/a.ts:8:6    [90mno-empty[39m                                 [33mEmpty block statement.[39m
[34mWarning[39m: /fixtures/a.ts:12:5 [90m@typescript-eslint/no-unused-vars[39m        [33m'x' is assigned a value but never used.[39m

[34m1 warning[39m
[31m4 errors[39m"
`;

exports[`autolinkable-stylish-formatter matches its snapshot with parsing errors 1`] = `
"[37m/fixtures/a.ts[39m
[31mError[39m: /fixtures/a.ts  [33mParsing error: If \\"parserOptions.project\\" has been set for @typescript-eslint/parser, /fixtures/a.ts must be included in at least one of the projects provided.[39m
[37m/fixtures/b.ts[39m
[31mError[39m: /fixtures/b.ts  [33mParsing error: If \\"parserOptions.project\\" has been set for @typescript-eslint/parser, /fixtures/b.ts must be included in at least one of the projects provided.[39m
[37m/fixtures/c.ts[39m
[31mError[39m: /fixtures/c.ts  [33mParsing error: If \\"parserOptions.project\\" has been set for @typescript-eslint/parser, /fixtures/c.ts must be included in at least one of the projects provided.[39m

[31m3 errors[39m"
`;
